File: 백준/Silver/15665. N과 M （11）/N과 M （11）.swift
var nums = readLine()!.split(separator: " ").map {Int(String($0))!}
var numbers = readLine()!.split(separator: " ").map {Int(String($0))!}
// 먼저 sorting하기
numbers.sort()
// 선택된 숫자를 넣는 배열
var selected = [String]()

func DFS(start: Int) {
    if selected.count == nums[1] {
        print(selected.joined(separator: " "))
        return
    }
    var temp = ""
    for num in 0..<nums[0] {
        if temp != String(numbers[num]) {
            selected.append(String(numbers[num]))
            temp = String(numbers[num])
            DFS(start: num+1)
            selected.removeLast()
        }
    }
}

DFS(start:0)

File: 백준/Silver/15665. N과 M （11）/N과 M （11）.py
n, m = map(int, input().split())
nums = list(map(int, input().split()))
nums.sort()

lst = []
visited = [False] * n
numSet = set()

def DFS(depth):
    if len(lst) == m:
        temp = " ".join(map(str, lst))
        if temp not in numSet:
            print(temp)
            numSet.add(temp)
        return
    for i in range(0, n):
        lst.append(nums[i])
        visited[i] = True
        DFS(i)
        lst.pop()
        visited[i] = False
DFS(0)